Authentication_and_Authorization/hello-security/signup-controller.php

```diff
@@ -1,19 +1,56 @@
 <?php
 include "db-connection.php";
 
-$conn = openCon();
-
-$sql = "INSERT INTO person (email, firstname, password, role) values ('"
-  . htmlentities($_POST["email"]) . "', '" 
-  . htmlentities($_POST["firstname"]). "', '"
-  . htmlentities($_POST["password"]) . "', '"
-  . htmlentities($_POST["role"]) . "');";
-// echo ("sql: " . $sql);  
-if (!$conn -> query($sql)) {
-  echo("Error description: " . $conn -> error);
+function checkIfEmailExist($email){
+  $conn = openCon();
+
+  $sql = "SELECT * FROM person where email = '" . $email . "';";
+  $preparedSql = $conn->prepare($sql);
+  $preparedSql->execute() or die("Failed to check is email exist.");
+  
+  $result = $preparedSql->fetchAll();
+  
+  return count($result) !== 0;
+}
+
+$email = htmlentities($_POST["email"]);
+$firstname = htmlentities($_POST["firstname"]);
+$password = htmlentities($_POST["password"]);
+$role = htmlentities($_POST["role"]);
+ 
+if( !empty($email) && !empty($firstname) && !empty($password) && !empty($role)){
+
+
+  if(strcmp($role, "admin")){
+
+    if(!checkIfEmailExist($email)){
+
+      $conn = openCon();
+      $hashedPassword = password_hash($password, PASSWORD_DEFAULT);
+
+      $sql = "INSERT INTO person (email, firstname, password, role) values ('"
+      . $email . "', '" 
+      . $firstname. "', '"
+      . $hashedPassword . "', '"
+      . $role . "');";
+
+      $preparedSql = $conn->prepare($sql) or die("Error description: " . $conn -> error);
+
+      $preparedSql->execute();
+      echo("You are registered as: " . $_POST["email"] . " with role: " .  $_POST["role"]);
+
+    } else {
+      echo("There is an account with this email!");
```

Comment:
You could redirect user to the signup page, if you want, so that the user can try again easily. This will save user's time.